editor/utils/url.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { parse, format } from 'url';
+
+/**
+ * Appends arguments to the query string of the url
+ *
+ * @param  {String} url   URL
+ * @param  {Object} args  Query Args
+ *
+ * @return {String}       Updated URL
+ */
+export function addQueryArgs( url, args ) {
+	const parsedUrl = parse( url, true );
+	const query = { ...parsedUrl.query, ...args };
```

Comment:
>Good point! that's my functional "everything is immutable" mind.

For sure, it's tough to balance with pragmatism toward performance. Usually I judge it based off of whether I created the object reference within scope of the same function. But yeah, in any case, the `delete`'s already there 😛